java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -2376,6 +2382,61 @@ public T objectRefcount(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Adds geospatial members with their positions to the specified sorted set stored at <code>key
+     * </code>.<br>
+     * If a member is already a part of the sorted set, its position is updated.
+     *
+     * @see <a href="https://redis.io/commands/geoadd/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param membersToGeoSpatialData A mapping of member names to their corresponding positions. See
+     *     {@link GeospatialData}. The command will report an error when the user attempts to index
+     *     coordinates outside the specified ranges.
+     * @param options The GeoAdd options. {@link GeoAddOptions}
+     * @return Command Response - The number of elements added to the sorted set. If <code>changed
+     *     </code> is set to <code>true</code> in the options, returns the number of elements updated
+     *     in the sorted set.
+     */
+    public T geoadd(
+            @NonNull String key,
+            @NonNull Map<String, GeospatialData> membersToGeoSpatialData,
+            @NonNull GeoAddOptions options) {
+        List<String> arguments = new ArrayList<>();
+        arguments.add(key);
+
+        if (options.getUpdateMode() != null) {
+            arguments.add(options.getUpdateMode().getRedisApi());
+        }
+
+        if (options.isChanged()) {
+            arguments.add("CH");
+        }
+
+        arguments.addAll(mapMemberToGeoDataToList(membersToGeoSpatialData));
+        ArgsArray commandArgs = buildArgs(arguments.toArray(new String[0]));
```

Comment:
I don't know actually, which one is better/faster
```suggestion
        ArgsArray commandArgs = buildArgs(arguments.toArray(String[]::new));
```

Reply:
From a quick glance does not seem like there is an immediate performance benefit.